stock_auto_move/models/stock_picking.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+
+
+class StockPicking(models.Model):
+
+    _inherit = 'stock.picking'
+
+    @api.model
+    def _transfer_pickings_with_auto_move(self, auto_moves_by_pickings):
+        """This function is meant to simulate what a user would normally
+        transfer a picking from the user interface either partial processing
+        or full processing.
+        @params auto_moves_by_pickings: dict of moves grouped by pickings
+        {stock.picking(id): stock.move(id1, id2, id3 ...), ...}
+        """
+        for picking in auto_moves_by_pickings:
+            if len(picking.move_lines) == len(auto_moves_by_pickings[picking]):
+                # Create immediate transfer wizard so it will fill the qty_done
+                # on the auto move linked operation
+                wizard = self.env['stock.immediate.transfer'].create(
+                    {'pick_id': picking.id})
+                wizard.process()
+            else:
+                # create back order with the auto moves
+                backorder = picking.copy(
```

Comment:
@zakiuu The workflow we discussed was quite right, but actually the good way is to create backorder with the rest of moves (like in standard).

For this, simply call the _create_backorder(backorder_moves) method on picking with backorder_moves = picking_moves - auto_run_moves (it will reassign the original picking and the operations should have been recreated in a good way)